contrib/nmbib/nmbibtex.1
.\" Automatically generated by Pod::Man 2.22 (Pod::Simple 3.07)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.ie \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.el \{\
.    de IX
..
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "NMBIBTEX 1"
.TH NMBIBTEX 1 "2015-04-25" "perl v5.10.1" "User Contributed Perl Documentation"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
nmbibtex \- a program to compile bibliographies for the package nmbib
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
nmbibtex \fI\s-1OPTIONS\s0\fR \fI\s-1FILE\s0\fR
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
nmbibtex calls \fIbibtex\fR\|(1) for all aux files produced by the package
nmbib
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-h\fR" 4
.IX Item "-h"
Print the help information and exit
.IP "\fB\-min\-crossrefs\fR \fInumber\fR" 4
.IX Item "-min-crossrefs number"
Send the option \f(CW\*(C`\-min\-crossref=NUMBER\*(C'\fR to \fIbibtex\fR\|(1)
.IP "\fB\-terse\fR" 4
.IX Item "-terse"
Send the option \f(CW\*(C`\-terse\*(C'\fR to \fIbibtex\fR\|(1)
.IP "\fB\-v\fR" 4
.IX Item "-v"
Print the copyright information and exit
.SH "AUTHORS AND LICENSE"
.IX Header "AUTHORS AND LICENSE"
Copyright 2015, Michael Cohen mcohen@u\-aizu.ac.jp
and Boris Veytsman borisv@lk.net
.PP
This work may be distributed and/or modified under the
conditions of the LaTeX Project Public License, either
version 1.3 of this license or (at your option) any 
later version.


contrib/nmbib/nmbibtex.pl
#!/usr/bin/env perl
# $Id: nmbibtex.pl,v 1.1 2015-04-26 00:19:37 boris Exp $
#
# Copyright 2015, Michael Cohen <mcohen@u-aizu.ac.jp>
# and Boris Veytsman <borisv@lk.net>
# This work may be distributed and/or modified under the
# conditions of the LaTeX Project Public License, either
# version 1.3 of this license or (at your option) any 
# later version.
# The latest version of the license is in
#    http://www.latex-project.org/lppl.txt
# and version 1.3 or later is part of all distributions of
# LaTeX version 2003/06/01 or later.
#
# This work has the LPPL maintenance status `maintained'.
#
# The Current Maintainer of this work is Boris Veytsman

=pod

=head1 NAME

nmbibtex - a program to compile bibliographies for the package nmbib

=head1 SYNOPSIS

nmbibtex I<OPTIONS> I<FILE>

=head1 DESCRIPTION

nmbibtex calls L<bibtex(1)> for all aux files produced by the package
nmbib

=head1 OPTIONS

=over 4

=item B<-h>

Print the help information and exit

=item B<-min-crossrefs> I<number>

Send the option C<-min-crossref=NUMBER> to L<bibtex(1)>

=item B<-terse>

Send the option C<-terse> to L<bibtex(1)>

=item B<-v>

Print the copyright information and exit

=back

=head1 AUTHORS AND LICENSE

Copyright 2015, Michael Cohen mcohen@u-aizu.ac.jp
and Boris Veytsman borisv@lk.net

This work may be distributed and/or modified under the
conditions of the LaTeX Project Public License, either
version 1.3 of this license or (at your option) any 
later version.

=cut

    use strict;

my $USAGE = <<'END';
Usage: nmbibtex [-min-crossref NUMBER] [-terse] FILE

Copyright 2015, Michael Cohen mcohen@u-aizu.ac.jp
and Boris Veytsman borisv@lk.net

This work may be distributed and/or modified under the
conditions of the LaTeX Project Public License, either
version 1.3 of this license or (at your option) any 
later version.
END

    use Getopt::Long;

my $help=0;
my $terse=0;
my $mincross=-1;

GetOptions("h" => \$help, 
	   'v' => \$help,
	   '-terse' => \$terse,
	   "-min-crossrefs=i" => \$mincross) or die $USAGE;

if ($help){ 
    die $USAGE;
}

my $cmdline = "bibtex ";
if ($terse) {
    $cmdline .= "-terse ";
}
if ($mincross>0) {
    $cmdline .= "-min-crossrefs=$mincross ";
}



my $file = shift;

if (!length($file)) {
    die $USAGE;
}

# Just in case strip the suffix
$file =~ s/\.aux$//;
$file =~ s/\.AUX$//;

my @auxfiles = glob("$file-*.aux");
foreach my $auxfile (@auxfiles) {
    print `$cmdline $auxfile`;
}
